profile/arc_cwl_ro_crate.md

```diff
@@ -1,23 +1,382 @@
+
+### CWL Workflow Run Profile
+
+The CWL Workflow Run Profile extends the [Workflow Run Crate](https://www.researchobject.org/workflow-run-crate/profiles/workflow_run_crate/). This profile describes the execution of a computational tool that orchestrates other tools, represented as a workflow executed using a Workflow Management System (WMS). The Workflow Run Crate combines [Process Run Crate](https://www.researchobject.org/workflow-run-crate/profiles/process_run_crate/)  and [Workflow RO-Crate](https://about.workflowhub.eu/Workflow-RO-Crate/), requiring a ComputationalWorkflow mainEntity and [CreateAction](https://schema.org/CreateAction) instances corresponding to the execution. Workflows can have multiple input and output parameters, defined optionally as FormalParameter entities and linked to the workflow's inputs and outputs.
+To continue staying consisten with the [ISA RO-Crate Profile](https://github.com/nfdi4plants/isa-ro-crate-profile/blob/main/profile/isa_ro_crate.md#isa-ro-crate-profile), we propose to add the [LabProcess](https://github.com/nfdi4plants/isa-ro-crate-profile/blob/main/profile/isa_ro_crate.md#labprocess) to the type CreateAction of the Process Run Crate within the Workflow Run Crate. This allows the annotation of inputs and outputs with metadata describing the properties of those Datasets and the processes leading from inputs to outputs.
 
 ```mermaid
 flowchart TD
-
+        A["File,\nSoftwareSourceCode,\nComputationalWorkflow,\nLabProtocol"] -- "input\noutput" --> B["FormalParameter"]
+        C["CreateAction,\nLabProcess"] -- "instrument" --> A
+        C -- "executes" --> A
+        C -- "agent" --> D["Person or Organization"]
+        B -- "exampleOfWork" --> E["File or Property Value"]
+        C -- "object result" --> E
+        D["Run=Dataset"] -- "processSequence=about" --> C
+        D -- "hasPart" --> E
 ```
+The `inputs` and `outputs` of the `ComputationalWorkflow` MAY point to the `objects` and `results` of `CreateAction` via `workExample`, while the latter point to the former via `exampleOfWork`.
 
 ## Requirements
 
+### CWL Workflow Profile
+
+The requirements of this profile are those of [Bioschemas ComputationalWorkflow Profile](https://bioschemas.org/profiles/ComputationalWorkflow/1.0-RELEASE#nav-description) 
+with the modifications listed below.
+
+#### ComputationalWorkflow
 | Property | Required | Expected Type | Description |
 |----------|----------|---------------|-------------|
-| | | | |
+| @type | MUST | [Text](https://schema.org/Text) | MUST be of type [File](https://schema.org/MediaObject), [SoftwareSourceCode](https://schema.org/SoftwareSourceCode), [ComputationalWorkflow](https://bioschemas.org/profiles/ComputationalWorkflow/1.0-RELEASE) and [LabProtocol](https://github.com/nfdi4plants/isa-ro-crate-profile/blob/main/profile/isa_ro_crate.md#labprotocol)|
```

Comment:
Regarding the `MUST ` multityping of File, SoftwareSourceCode and ComputationalWorkflow:

I know this stems from the [workflow-run-crate-profile](https://www.researchobject.org/workflow-run-crate/profiles/workflow_run_crate/), but there I could only find it specified in this way in the graph.

Wouldn't this logically make more sense to be an `OR` between these types instead of an `AND`? Like in, either it's a tool execution or a workflow execution?

Maybe this isn't the right place here to discuss this, but just noticed it.